devito/ir/clusters/algorithms.py

```diff
@@ -213,16 +248,16 @@ def bump_and_contract(targets, source, sink):
             processed.append(e.func(e.lhs, e.rhs.xreplace(mapper)))
         else:
             for i in sink.tensors[function]:
-                scalarized = Scalar(name='s%d' % len(mapper)).indexify()
-                mapper[i] = scalarized
+                scalar = Scalar(name='s%s%d' % (i.function.name, len(mapper))).indexify()
```

Comment:
what was the issue here?

Reply:
In some (very odd) cases, there was some duplicates temporaries so the compiler was complaining about redeclaration.

Reply:
do we have a test covering this? but yes, I can see why this might happen. This is a symptom that this routine should better be moved to the DSE... (of course, this is on me and for another day)